### ISSUE TITLE
coss create_order TypeError: string indices must be integers

### ISSUE THREAD

**Issue by AlexVas1** [ccxt/ccxt#4612]
- OS: WIN7
- Programming Language version: Python 3.6.5
- CCXT version: 1.18.201
- Exchange: coss
- Method: create_order

Traceback (most recent call last):
  File "C:\temp\python\coss_auth.py", line 35, in <module>
    ordernew = coss.create_order('ETH/BTC','limit','SELL',9885,111)
  File "C:\Users\Пользователь\AppData\Local\Programs\Python\Python36-32\lib\site-packages\ccxt\coss.py", line 829, in create_order
    return self.parse_order(response, market)
  File "C:\Users\Пользователь\AppData\Local\Programs\Python\Python36-32\lib\site-packages\ccxt\coss.py", line 768, in parse_order
    type = self.safe_string(order, 'type')
  File "C:\Users\Пользователь\AppData\Local\Programs\Python\Python36-32\lib\site-packages\ccxt\base\exchange.py", line 550, in safe_string
    return str(dictionary[key]) if key is not None and (key in dictionary) and dictionary[key] is not None else default_value
TypeError: string indices must be integers

Why?

**Comment by kroitor** [comment]
```Python
ordernew = coss.create_order('ETH/BTC','limit','SELL',9885,111)  # wrong
# should be a lowercase 'buy' or 'sell' string  ↑↓ according to the Manual
ordernew = coss.create_order('ETH/BTC','limit','sell',9885,111)  # correct
```

Let us know if that doesn't help.

**Comment by kroitor** [comment]
```shell
mbp:ccxt igorkroitor$ python examples/py/cli.py coss fetch_ticker ETH/BTC
```

```Python
{'ask': 0.03097,  # ←----------------------------- current ask price is 0.03097 BTC for one ETH
 'askVolume': None,
 'average': None,
 'baseVolume': 17.2264738372093,
 'bid': 0.03095999,  # ←----------------------- current bid price is 0.03095999 BTC for one ETH
 'bidVolume': None,
 'change': 0.0,
 'close': 0.03096,
 'datetime': '2019-02-06T00:00:02.163Z',
 'high': 0.03136,
 'info': {u'Ask': u'0.03097000',
          u'BaseVolume': 0.53333163,
          u'Bid': u'0.03095999',
          u'High': 0.03136,
          u'Last': 0.03096,
          u'Low': 0.03087,
          u'MarketName': u'ETH-BTC',
          u'PrevDay': 0.03096,
          u'TimeStamp': u'2019-02-06T00:00:02.163Z',
          u'Volume': 17.2264738372093},
 'last': 0.03096,
 'low': 0.03087,
 'open': 0.03096,
 'percentage': 0.0,
 'previousClose': None,
 'quoteVolume': 0.53333163,
 'symbol': u'ETH/BTC',
 'timestamp': 1549411202163,
 'vwap': None}
mbp:ccxt igorkroitor$
```

**Comment by kroitor** [comment]
```Python
ordernew = coss.create_order(type='limit',symbol='ETH/BTC',price=166.98,amount=0.1,side='BUY',params={})
#                                                                     this is still wrong ↑
```

**Comment by AlexVas1** [comment]
example from manual
---------
symbol = 'ETH/BTC'
type = 'limit'  # or 'market', other types aren't unified yet
side = 'sell'
amount = 123.45  # your amount
price = 54.321  # your price
# overrides
params = {
    'stopPrice': 123.45,  # your stop price
    'type': 'stopLimit',
}
order = coss.create_order(symbol, type, side, amount, price, params)
#order = coss.create_order(symbol, type, side, amount, price, params)
print(order)
---------------
Traceback (most recent call last):
  File "C:\temp\python\coss_auth.py", line 48, in <module>
    order = coss.create_order(symbol, type, side, amount, price, params)
  File "C:\Users\Пользователь\AppData\Local\Programs\Python\Python36-32\lib\site-packages\ccxt\coss.py", line 829, in create_order
    return self.parse_order(response, market)
  File "C:\Users\Пользователь\AppData\Local\Programs\Python\Python36-32\lib\site-packages\ccxt\coss.py", line 768, in parse_order
    type = self.safe_string(order, 'type')
  File "C:\Users\Пользователь\AppData\Local\Programs\Python\Python36-32\lib\site-packages\ccxt\base\exchange.py", line 550, in safe_string
    return str(dictionary[key]) if key is not None and (key in dictionary) and dictionary[key] is not None else default_value
TypeError: string indices must be integers

**Comment by frosty00** [comment]
can you make `coss.verbose = True` and paste the output here

**Comment by AlexVas1** [comment]
coss = ccxt.coss({
        'apiKey': "#########",
        'secret': "######"
    })

balance request works fine

**Comment by kroitor** [comment]
That's an pseudocode example of how to override exchange-specific `params` for special cases, not what you really want – it wouldn't even work with COSS, I'm afraid. In order to work with the lib efficiently and understand the purpose of `params` and how to place orders properly, you need to read the entire Manual from the beginning to the end, no shortcuts, you can't just copy any snippet and expect it to work for your use-case as is without your involvement, unfortunately. The examples are meant for demonstration of certain concepts, and if you're using `params` make sure you understand what they are for. Otherwise don't use them. COSS doesn't support the `stopLimit` override at all.

**Comment by AlexVas1** [comment]
my first example - without params

**Comment by kroitor** [comment]
> my first example - without params
but with tons of other errors, contradicting the Manual

**Comment by AlexVas1** [comment]
new example from manual without params
----
symbol = 'ETH/BTC'
type = 'limit'  # or 'market', other types aren't unified yet
side = 'sell'
amount = 123.45  # your amount
price = 54.321  # your price
# overrides
params = {
    'stopPrice': 123.45,  # your stop price
    'type': 'stopLimit',
}
order = coss.create_order(symbol, type, side, amount, price)
print(order)
-----

Traceback (most recent call last):
  File "C:\temp\python\coss_auth.py", line 49, in <module>
    order = coss.create_order(symbol, type, side, amount, price)
  File "C:\Users\Пользователь\AppData\Local\Programs\Python\Python36-32\lib\site-packages\ccxt\coss.py", line 829, in create_order
    return self.parse_order(response, market)
  File "C:\Users\Пользователь\AppData\Local\Programs\Python\Python36-32\lib\site-packages\ccxt\coss.py", line 768, in parse_order
    type = self.safe_string(order, 'type')
  File "C:\Users\Пользователь\AppData\Local\Programs\Python\Python36-32\lib\site-packages\ccxt\base\exchange.py", line 550, in safe_string
    return str(dictionary[key]) if key is not None and (key in dictionary) and dictionary[key] is not None else default_value
TypeError: string indices must be integers

**Comment by AlexVas1** [comment]
another one example from manual with empty params:
symbol = 'ETH/BTC'
type = 'limit'  # or 'market', other types aren't unified yet
side = 'sell'
amount = 123.45  # your amount
price = 54.321  # your price
# overrides
params = {}
order = coss.create_order(symbol, type, side, amount, price, params)
print(order)

answer:
Traceback (most recent call last):
  File "C:\temp\python\coss_auth.py", line 46, in <module>
    order = coss.create_order(symbol, type, side, amount, price, params)
  File "C:\Users\Пользователь\AppData\Local\Programs\Python\Python36-32\lib\site-packages\ccxt\coss.py", line 829, in create_order
    return self.parse_order(response, market)
  File "C:\Users\Пользователь\AppData\Local\Programs\Python\Python36-32\lib\site-packages\ccxt\coss.py", line 768, in parse_order
    type = self.safe_string(order, 'type')
  File "C:\Users\Пользователь\AppData\Local\Programs\Python\Python36-32\lib\site-packages\ccxt\base\exchange.py", line 550, in safe_string
    return str(dictionary[key]) if key is not None and (key in dictionary) and dictionary[key] is not None else default_value
TypeError: string indices must be integers

**Comment by kroitor** [comment]
Add your keys, a correct price and amount to the following and run, then copy-paste the entire output here, please use triple backticks (&#096;&#096;&#096;code&#096;&#096;&#096;) to format the code for readability as describe here: https://github.com/ccxt/ccxt/blob/master/CONTRIBUTING.md#how-to-submit-an-issue

```Python
import ccxt
exchange = ccxt.coss({
    'apiKey': 'YOUR_API_KEY',  # ←----------------- put your values here
    'secret': 'YOUR_SECRET',  # ←----------------- put your values here
    'enableRateLimit': True,  # required by the Manual
})
exchange.load_markets()
exchange.verbose = True

exchange.fetch_balance()

symbol = 'ETH/BTC'
amount = YOUR_CORRECT_AMOUNT  # float   ←----------------- put your values here
price = YOUR_CORRECT_PRICE  # float   ←----------------- put your values here
side = 'sell'  # or 'buy'   # ←----------------- put your values here
type = 'limit'

order = exchange.create_order(symbol, type, side, amount, price)
```

**Comment by AlexVas1** [comment]
'''import ccxt 

exchange = ccxt.coss({
        'apiKey': "#####",
        'secret': "###",
    'enableRateLimit': True,  # required by the Manual
})
exchange.load_markets()
exchange.verbose = True  # ←--------------- this is needed here

exchange.fetch_balance()

symbol = 'ETH/BTC'
amount = 0.006  # float   ←----------------- put your values here
price = 0.03096  # float   ←----------------- put your values here
side = 'sell'  # or 'buy'   # ←----------------- put your values here
type = 'limit'

order = exchange.create_order(symbol, type, side, amount, price)'''

Request: GET https://trade.coss.io/c/api/v1/account/balances?timestamp=1549415123880 {'Signature': '86123b09487a00502fd8d6cf0181f0d1afe21dc4ad936f7f6d0b37ddfb5594d9', 'Authorization': '----cut---', 'User-Agent': 'python-requests/2.21.0', 'Accept-Encoding': 'gzip, deflate'} None

Response: GET https://trade.coss.io/c/api/v1/account/balances?timestamp=1549415123880 200 {'Server': 'Apache-Coyote/1.1', 'X-Frame-Options': 'SAMEORIGIN', 'X-Content-Type-Options': 'nosniff', 'Content-Type': 'application/json;charset=UTF-8', 'Transfer-Encoding': 'chunked', 'Date': 'Wed, 06 Feb 2019 01:05:23 GMT', 'Via': '1.1 google', 'Alt-Svc': 'clear'} [{"currency_code":"XEM","address":"","total":"0","available":"0","in_order":"0","memo":null},{"currency_code":"ZEN","address":"","total":"0","available":"0","in_order":"0","memo":null},{"currency_code":"VET","address":"","total":"0","available":"0","in_order":"0","memo":null},{"currency_code":"ADI","address":"","total":"0","available":"0","in_order":"0","memo":null},{"currency_code":"WTC","address":"","total":"0","available":"0","in_order":"0","memo":null},{"currency_code":"ITT","address":"","total":"0","available":"0","in_order":"0","memo":null},{"currency_code":"BNT","address":"","total":"0","available":"0","in_order":"0","memo":null},{"currency_code":"LALA","address":"","total":"0","available":"0","in_order":"0","memo":null},{"currency_code":"PAY","address":"","total":"0","available":"0","in_order":"0","memo":null},{"currency_code":"USDT","address":"","total":"0","available":"0","in_order":"0","memo":null},{"currency_code":"NPXSXEM","address":"","total":"0","available":"0","in_order":"0","memo":null},{"currency_code":"DASH","address":"","total":"0","available":"0","in_order":"0","memo":null},{"currency_code":"TRAK","address":"","total":"0","available":"0","in_order":"0","memo":null},{"currency_code":"WISH","address":"","total":"0","available":"0","in_order":"0","memo":null},{"currency_code":"VZT","address":"","total":"0","available":"0","in_order":"0","memo":null},{"currency_code":"OPQ","address":"","total":"0","available":"0","in_order":"0","memo":null},{"currency_code":"WAVES","address":"","total":"0","available":"0","in_order":"0","memo":null},{"currency_code":"PRIX","address":"","total":"0","available":"0","in_order":"0","memo":null},{"currency_code":"XNK","address":"","total":"0","available":"0","in_order":"0","memo":null},{"currency_code":"STX","address":"","total":"0","available":"0","in_order":"0","memo":null},{"currency_code":"XRP","address":null,"total":"0","available":"0","in_order":"0","memo":null},{"currency_code":"NPXS","address":"0xbae9d46f74c440b21d0d59ad3160680a7621facd","total":"0","available":"0","in_order":"0","memo":null},{"currency_code":"TRX","address":"","total":"0","available":"0","in_order":"0","memo":null},{"currency_code":"BWT","address":"0xbae9d46f74c440b21d0d59ad3160680a7621facd","total":"0.32257","available":"0.32257","in_order":"0","memo":null},{"currency_code":"GUSD","address":"","total":"0","available":"0","in_order":"0","memo":null},{"currency_code":"SENC","address":"","total":"0","available":"0","in_order":"0","memo":null},{"currency_code":"SUB","address":"","total":"0","available":"0","in_order":"0","memo":null},{"currency_code":"EUR","address":"","total":"0","available":"0","in_order":"0","memo":null},{"currency_code":"LSK","address":"","total":"0","available":"0","in_order":"0","memo":null},{"currency_code":"BCH","address":"","total":"0","available":"0","in_order":"0","memo":null},{"currency_code":"POE","address":"","total":"0","available":"0","in_order":"0","memo":null},{"currency_code":"OMG","address":"","total":"0","available":"0","in_order":"0","memo":null},{"currency_code":"BTC","address":"1NW9GpehUaDvnAouCKQko6BzGbUmQHbLxR","total":"0.00064008","available":"0.00064008","in_order":"0","memo":null},{"currency_code":"CAN","address":"","total":"0","available":"0","in_order":"0","memo":null},{"currency_code":"CVC","address":"","total":"0","available":"0","in_order":"0","memo":null},{"currency_code":"KIN","address":"","total":"0","available":"0","in_order":"0","memo":null},{"currency_code":"SURE","address":"","total":"0","available":"0","in_order":"0","memo":null},{"currency_code":"HGT","address":"","total":"0","available":"0","in_order":"0","memo":null},{"currency_code":"PASS","address":"","total":"0","available":"0","in_order":"0","memo":null},{"currency_code":"TUSD","address":"","total":"0","available":"0","in_order":"0","memo":null},{"currency_code":"NOX","address":"","total":"0","available":"0","in_order":"0","memo":null},{"currency_code":"USDC","address":"","total":"0","available":"0","in_order":"0","memo":null},{"currency_code":"ARK","address":"","total":"0","available":"0","in_order":"0","memo":null},{"currency_code":"LTC","address":"","total":"0","available":"0","in_order":"0","memo":null},{"currency_code":"GAT","address":"","total":"0","available":"0","in_order":"0","memo":null},{"currency_code":"PGT","address":"","total":"0","available":"0","in_order":"0","memo":null},{"currency_code":"IND","address":"","total":"0","available":"0","in_order":"0","memo":null},{"currency_code":"ENJ","address":"","total":"0","available":"0","in_order":"0","memo":null},{"currency_code":"KNC","address":"","total":"0","available":"0","in_order":"0","memo":null},{"currency_code":"FXT","address":"","total":"0","available":"0","in_order":"0","memo":null},{"currency_code":"MRK","address":"","total":"0","available":"0","in_order":"0","memo":null},{"currency_code":"CFT","address":"","total":"0","available":"0","in_order":"0","memo":null},{"currency_code":"BLZ","address":"","total":"0","available":"0","in_order":"0","memo":null},{"currency_code":"CRED","address":"","total":"0","available":"0","in_order":"0","memo":null},{"currency_code":"COSS","address":"","total":"0","available":"0","in_order":"0","memo":null},{"currency_code":"SNM","address":"","total":"0","available":"0","in_order":"0","memo":null},{"currency_code":"GBP","address":"","total":"0","available":"0","in_order":"0","memo":null},{"currency_code":"UBC","address":"","total":"0","available":"0","in_order":"0","memo":null},{"currency_code":"XLM","address":null,"total":"0","available":"0","in_order":"0","memo":null},{"currency_code":"LINK","address":"","total":"0","available":"0","in_order":"0","memo":null},{"currency_code":"FYN","address":"","total":"0","available":"0","in_order":"0","memo":null},{"currency_code":"FDX","address":"","total":"0","available":"0","in_order":"0","memo":null},{"currency_code":"QTUM","address":"","total":"0","available":"0","in_order":"0","memo":null},{"currency_code":"UFR","address":"","total":"0","available":"0","in_order":"0","memo":null},{"currency_code":"XDCE","address":"","total":"0","available":"0","in_order":"0","memo":null},{"currency_code":"USD","address":"CLH7VGDBKHSS","total":"0","available":"0","in_order":"0","memo":null},{"currency_code":"EOS","address":null,"total":"0","available":"0","in_order":"0","memo":null},{"currency_code":"CS","address":"","total":"0","available":"0","in_order":"0","memo":null},{"currency_code":"MORE","address":"","total":"0","available":"0","in_order":"0","memo":null},{"currency_code":"TIG","address":"","total":"0","available":"0","in_order":"0","memo":null},{"currency_code":"LA","address":"","total":"0","available":"0","in_order":"0","memo":null},{"currency_code":"DAT","address":"","total":"0","available":"0","in_order":"0","memo":null},{"currency_code":"JET","address":"","total":"0","available":"0","in_order":"0","memo":null},{"currency_code":"LAN","address":"","total":"0","available":"0","in_order":"0","memo":null},{"currency_code":"ETH","address":"0xbae9d46f74c440b21d0d59ad3160680a7621facd","total":"0.000961","available":"0.000961","in_order":"0","memo":null},{"currency_code":"PRSN","address":"","total":"0","available":"0","in_order":"0","memo":null},{"currency_code":"NEO","address":"","total":"0","available":"0","in_order":"0","memo":null},{"currency_code":"KAYA","address":"","total":"0","available":"0","in_order":"0","memo":null},{"currency_code":"PIX","address":"","total":"0","available":"0","in_order":"0","memo":null},{"currency_code":"REQ","address":"","total":"0","available":"0","in_order":"0","memo":null},{"currency_code":"ICX","address":"","total":"0","available":"0","in_order":"0","memo":null}]

Request: POST https://trade.coss.io/c/api/v1/order/add {'Signature': '8bc8f8bc6096fb81db46e56ecfb55deb824c521217a70650fa3a45066b067627', 'Authorization': '----cut---', 'User-Agent': 'python-requests/2.21.0', 'Accept-Encoding': 'gzip, deflate'} {"timestamp":1549415124880,"order_symbol":"ETH_BTC","order_price":0.03096,"order_size":0.006,"order_side":"SELL","type":"limit"}

Response: POST https://trade.coss.io/c/api/v1/order/add 200 {'Server': 'Apache-Coyote/1.1', 'X-Frame-Options': 'SAMEORIGIN', 'X-Content-Type-Options': 'nosniff', 'Set-Cookie': 'JSESSIONID=67D7E554B2D3D65A3C4B59F909AA88DF; Path=/c/; HttpOnly', 'Content-Type': 'text/html;charset=UTF-8', 'Content-Language': 'en', 'Content-Encoding': 'gzip', 'Vary': 'Accept-Encoding', 'Date': 'Wed, 06 Feb 2019 01:05:25 GMT', 'Via': '1.1 google', 'Alt-Svc': 'clear', 'Transfer-Encoding': 'chunked'} 












<!DOCTYPE html>
<html>
  <head>
	
    
    <meta charset="utf-8">
<meta name="description" content="Welcome to COSS           a fast, secure, and simple exchange supporting a multitude of cryptocurrencies           What does COSS mean?           COSS stands for Crypto One-Stop Solution. We are your one-stop shop for all things crypto: an exchange, an e-wallet which supports a broad variety of tokens, a platform for ICO launches and promotional trading campaigns, a fiat gateway, a market cap widget, and more.">
<meta name="keywords" content="coss coin crypto exchange, crypto one-stop solution, coss exchange, coss.io, coss coin, coss exchange home, coss.io home, coss crypto exchange, cryptocurrency exchange coss, cryptocurrency list coss, cryptocoin exchange coss, coss cryptocurrency list, crypto exchange list, coss fee split allocation, ETH, Ethereum, ethereum blockchain, ERC20, ERC223, cryptotoken, cryptocurrency, coss coin payouts, coss coin dividends, coss coin revenue, crypto coin exchange coss, crypto token coss, cosscrypto, coss crypto, coss coinmarket cap, coss coin market, coss io marketcap, coss token list, coss token exchange, coin listing coss, coss coin value, coss btc, coss io fees, buy coss, coss token price, coin market cap coss, coss io price, coss token, coss io coin, coss cryptocurrency, crypto listing coss, cryptocurrency coin list coss, coss platform coins list, crypto listings coss, coss coin exchange list, buy coss coin, coss exchange fees, coss exchange fee split, coss crypto price, how to buy coss, how to buy coss coin, coss cryptocurrency, where to buy coss, coss coin wallet, coss io exchange, where to buy coss, where to buy coss coin, coss crypto exchange fee, list your coin coss, list cryptocurrency coss, crypto platform coss, coss crypto exchange">
<meta http-equiv="x-ua-compatible" content="ie=edge">

<meta name='viewport' content='width=device-width, initial-scale=1.0, maximum-scale=1.0, user-scalable=0' />

<script type="text/javascript">window.NREUM||(NREUM={}),__nr_require=function(e,n,t){function r(t){if(!n[t]){var o=n[t]={exports:{}};e[t][0].call(o.exports,function(n){var o=e[t][1][n];return r(o||n)},o,o.exports)}return n[t].exports}if("function"==typeof __nr_require)return __nr_require;for(var o=0;o<t.length;o++)r(t[o]);return r}({1:[function(e,n,t){function r(){}function o(e,n,t){return function(){return i(e,[c.now()].concat(u(arguments)),n?null:this,t),n?void 0:this}}var i=e("handle"),a=e(3),u=e(4),f=e("ee").get("tracer"),c=e("loader"),s=NREUM;"undefined"==typeof window.newrelic&&(newrelic=s);var p=["setPageViewName","setCustomAttribute","setErrorHandler","finished","addToTrace","inlineHit","addRelease"],d="api-",l=d+"ixn-";a(p,function(e,n){s[n]=o(d+n,!0,"api")}),s.addPageAction=o(d+"addPageAction",!0),s.setCurrentRouteName=o(d+"routeName",!0),n.exports=newrelic,s.interaction=function(){return(new r).get()};var m=r.prototype={createTracer:function(e,n){var t={},r=this,o="function"==typeof n;return i(l+"tracer",[c.now(),e,t],r),function(){if(f.emit((o?"":"no-")+"fn-start",[c.now(),r,o],t),o)try{return n.apply(this,arguments)}catch(e){throw f.emit("fn-err",[arguments,this,e],t),e}finally{f.emit("fn-end",[c.now()],t)}}}};a("actionText,setName,setAttribute,save,ignore,onEnd,getContext,end,get".split(","),function(e,n){m[n]=o(l+n)}),newrelic.noticeError=function(e,n){"string"==typeof e&&(e=new Error(e)),i("err",[e,c.now(),!1,n])}},{}],2:[function(e,n,t){function r(e,n){if(!o)return!1;if(e!==o)return!1;if(!n)return!0;if(!i)return!1;for(var t=i.split("."),r=n.split("."),a=0;a<r.length;a++)if(r[a]!==t[a])return!1;return!0}var o=null,i=null,a=/Version\/(\S+)\s+Safari/;if(navigator.userAgent){var u=navigator.userAgent,f=u.match(a);f&&u.indexOf("Chrome")===-1&&u.indexOf("Chromium")===-1&&(o="Safari",i=f[1])}n.exports={agent:o,version:i,match:r}},{}],3:[function(e,n,t){function r(e,n){var t=[],r="",i=0;for(r in e)o.call(e,r)&&(t[i]=n(r,e[r]),i+=1);return t}var o=Object.prototype.hasOwnProperty;n.exports=r},{}],4:[function(e,n,t){function r(e,n,t){n||(n=0),"undefined"==typeof t&&(t=e?e.length:0);for(var r=-1,o=t-n||0,i=Array(o<0?0:o);++r<o;)i[r]=e[n+r];return i}n.exports=r},{}],5:[function(e,n,t){n.exports={exists:"undefined"!=typeof window.performance&&window.performance.timing&&"undefined"!=typeof window.performance.timing.navigationStart}},{}],ee:[function(e,n,t){function r(){}function o(e){function n(e){return e&&e instanceof r?e:e?f(e,u,i):i()}function t(t,r,o,i){if(!d.aborted||i){e&&e(t,r,o);for(var a=n(o),u=v(t),f=u.length,c=0;c<f;c++)u[c].apply(a,r);var p=s[y[t]];return p&&p.push([b,t,r,a]),a}}function l(e,n){h[e]=v(e).concat(n)}function m(e,n){var t=h[e];if(t)for(var r=0;r<t.length;r++)t[r]===n&&t.splice(r,1)}function v(e){return h[e]||[]}function g(e){return p[e]=p[e]||o(t)}function w(e,n){c(e,function(e,t){n=n||"feature",y[t]=n,n in s||(s[n]=[])})}var h={},y={},b={on:l,addEventListener:l,removeEventListener:m,emit:t,get:g,listeners:v,context:n,buffer:w,abort:a,aborted:!1};return b}function i(){return new r}function a(){(s.api||s.feature)&&(d.aborted=!0,s=d.backlog={})}var u="nr@context",f=e("gos"),c=e(3),s={},p={},d=n.exports=o();d.backlog=s},{}],gos:[function(e,n,t){function r(e,n,t){if(o.call(e,n))return e[n];var r=t();if(Object.defineProperty&&Object.keys)try{return Object.defineProperty(e,n,{value:r,writable:!0,enumerable:!1}),r}catch(i){}return e[n]=r,r}var o=Object.prototype.hasOwnProperty;n.exports=r},{}],handle:[function(e,n,t){function r(e,n,t,r){o.buffer([e],r),o.emit(e,n,t)}var o=e("ee").get("handle");n.exports=r,r.ee=o},{}],id:[function(e,n,t){function r(e){var n=typeof e;return!e||"object"!==n&&"function"!==n?-1:e===window?0:a(e,i,function(){return o++})}var o=1,i="nr@id",a=e("gos");n.exports=r},{}],loader:[function(e,n,t){function r(){if(!E++){var e=x.info=NREUM.info,n=l.getElementsByTagName("script")[0];if(setTimeout(s.abort,3e4),!(e&&e.licenseKey&&e.applicationID&&n))return s.abort();c(y,function(n,t){e[n]||(e[n]=t)}),f("mark",["onload",a()+x.offset],null,"api");var t=l.createElement("script");t.src="https://"+e.agent,n.parentNode.insertBefore(t,n)}}function o(){"complete"===l.readyState&&i()}function i(){f("mark",["domContent",a()+x.offset],null,"api")}function a(){return O.exists&&performance.now?Math.round(performance.now()):(u=Math.max((new Date).getTime(),u))-x.offset}var u=(new Date).getTime(),f=e("handle"),c=e(3),s=e("ee"),p=e(2),d=window,l=d.document,m="addEventListener",v="attachEvent",g=d.XMLHttpRequest,w=g&&g.prototype;NREUM.o={ST:setTimeout,SI:d.setImmediate,CT:clearTimeout,XHR:g,REQ:d.Request,EV:d.Event,PR:d.Promise,MO:d.MutationObserver};var h=""+location,y={beacon:"bam.nr-data.net",errorBeacon:"bam.nr-data.net",agent:"js-agent.newrelic.com/nr-1118.min.js"},b=g&&w&&w[m]&&!/CriOS/.test(navigator.userAgent),x=n.exports={offset:u,now:a,origin:h,features:{},xhrWrappable:b,userAgent:p};e(1),l[m]?(l[m]("DOMContentLoaded",i,!1),d[m]("load",r,!1)):(l[v]("onreadystatechange",o),d[v]("onload",r)),f("mark",["firstbyte",u],null,"api");var E=0,O=e(5)},{}]},{},["loader"]);</script><link rel="shortcut icon" href="/static/imgs/android-icon-192x192.png" type="image/x-icon">

<link rel="icon" type="image/png" sizes="192x192" href="/static/imgs/android-icon-192x192.png">
<link rel="icon" type="image/png" sizes="32x32" href="/static/imgs/favicon-32x32.png">
<link rel="icon" type="image/png" sizes="96x96" href="/static/imgs/favicon-96x96.png">
<link rel="icon" type="image/png" sizes="16x16" href="/static/imgs/favicon-16x16.png">
<title>Crypto One-Stop Solution platform - COSS Exchange homepage</title>

<script type="text/javascript" src="//translate.google.com/translate_a/element.js?cb=googleTranslateElementInit" async></script>
    <link rel="shortcut icon" href="/c/resources/vueapp/dist/static/imgs/android-icon-192x192-7bc929141bb1c59c53e9802fc009ebeb.png" type="image/x-icon">

    <link rel="icon" type="image/png" sizes="192x192" href="/c/resources/vueapp/dist/static/imgs/android-icon-192x192-7bc929141bb1c59c53e9802fc009ebeb.png">
    <link rel="icon" type="image/png" sizes="32x32" href="/c/resources/vueapp/dist/static/imgs/favicon-32x32-53dd6e6d6d25a4097becce510a1bae81.png">
    <link rel="icon" type="image/png" sizes="96x96" href="/c/resources/vueapp/dist/static/imgs/favicon-96x96-7e556bf30d2716ca3556083bf88a8978.png">
    <link rel="icon" type="image/png" sizes="16x16" href="/c/resources/vueapp/dist/static/imgs/favicon-16x16-cdc511534b3dfdf9b82d518d4deb7922.png">
		<link href=/c/resources/vueapp/dist/static/css/app.a83b6ced8e19cd1cf7b1b69c07e0e96f.css rel=stylesheet>
  </head>
  <body>
    <div id="app"></div>
		
		







<script>
  (function() {
    var accountPhone = '';
    var csrfname = '';
    var csrftoken = '';
    var accountCountry = 'US';
    var accountLocale = 'en' || '';
    var isLoggedIn = 'false' === 'false' ? false : true;
    accountLocale = accountLocale.toLowerCase();

    if (accountLocale === 'vn') accountLocale = 'vi';
    else if (accountLocale === 'us') accountLocale = 'en';
    
    var user = {
      locale: accountLocale,
      isLoggedIn: isLoggedIn,
      country: accountCountry,
      phone: accountPhone,
      csrfname: csrfname,
      csrftoken: csrftoken,
    };

    window.user = user;
  })();
</script>
    <script src=/c/resources/vueapp/dist/static/js/manifest.6cb1c23157d4d1a0d411.js></script><script src=/c/resources/vueapp/dist/static/js/vendor.180783bd722e805aa997.js></script><script src=/c/resources/vueapp/dist/static/js/app.1521d84fd92a4d4f5279.js></script>
  
<script type="text/javascript">window.NREUM||(NREUM={});NREUM.info={"errorBeacon":"bam.nr-data.net","licenseKey":"6351711e1c","agent":"","beacon":"bam.nr-data.net","applicationTime":12,"applicationID":"128798597","transactionName":"b----cut---==","queueTime":0}</script></body>
</html>
Traceback (most recent call last):
  File "C:\temp\python\coss_auth.py", line 29, in <module>
    order = exchange.create_order(symbol, type, side, amount, price)
  File "C:\Users\Пользователь\AppData\Local\Programs\Python\Python36-32\lib\site-packages\ccxt\coss.py", line 829, in create_order
    return self.parse_order(response, market)
  File "C:\Users\Пользователь\AppData\Local\Programs\Python\Python36-32\lib\site-packages\ccxt\coss.py", line 768, in parse_order
    type = self.safe_string(order, 'type')
  File "C:\Users\Пользователь\AppData\Local\Programs\Python\Python36-32\lib\site-packages\ccxt\base\exchange.py", line 550, in safe_string
    return str(dictionary[key]) if key is not None and (key in dictionary) and dictionary[key] is not None else default_value
TypeError: string indices must be integers
>>>

**Comment by kroitor** [comment]
@AlexVas1 the code is correct now, however, your balance is zero, so, it's insufficient to place the order. Under regular circumstances the exchange class would throw an `InsufficientBalance` exception as described in the Manual for you to be able to handle that. However, in case of the COSS exchange, the error reporting is apparently not working right, so we have reached out to the COSS tech team on this matter earlier. It will work with proper balances, however, in case of an error it can break down like that. We will fix this as soon as we have an update from COSS on this issue and will let you know.
